oaebu_workflows/workflows/thoth_telescope.py

```diff
@@ -0,0 +1,331 @@
+        success = upload_files_to_cloud_storage(self.download_bucket, [blob], download_files)
+        if not success:
+            raise AirflowException("Blob could not be uploaded to cloud storage")
+
+    def transform(self, release: ThothRelease, **kwargs) -> None:
+        """Task to transform the Thoth ONIX data"""
+        parse_onix(self.download_folder, self.transform_folder, make_org_id(self.organisation_name))
+        # Rename file
+        shutil.move(
+            os.path.join(self.transform_folder, "full.jsonl"),
+            os.path.join(self.transform_folder, self.transform_file_name),
+        )
+
+    def upload_transformed(self, release: ThothRelease, **kwargs) -> None:
+        """Upload the downloaded thoth onix .jsonl to google cloud bucket"""
+        transform_files = list_files(self.transform_folder, self.transform_file_name)
+        if len(transform_files) != 1:
+            raise AirflowException(
+                f"Unexpected number of files in transform folder. Expected 1, found {len(transform_files)}"
+            )
+        blob = blob_name_from_path(transform_files[0])
+        success = upload_files_to_cloud_storage(self.transform_bucket, [blob], transform_files)
+        if not success:
+            raise AirflowException("Blob could not be uploaded to cloud storage")
+
+    def bq_load(self, release: ThothRelease, **kwargs) -> None:
+        """Task to load the transformed ONIX jsonl file to BigQuery.
+
+        The table_id is set to the file name without the extension.
+        """
+        transform_files = list_files(self.transform_folder, self.transform_file_name)
+        if len(transform_files) != 1:
+            raise AirflowException(
+                f"Unexpected number of files in transform folder. Expected 1, found {len(transform_files)}"
+            )
+        # Load each transformed release
+        transform_blob = blob_name_from_path(transform_files[0])
+        table_id, _ = table_ids_from_path(transform_files[0])
+        schema_file_path = find_schema(path=self.schema_folder, table_name="onix")
+        bq_load_shard(
+            schema_file_path=schema_file_path,
+            project_id=self.project_id,
+            transform_bucket=self.transform_bucket,
+            transform_blob=transform_blob,
+            dataset_id=self.dataset_id,
+            data_location=self.data_location,
+            table_id=table_id,
+            release_date=release.release_date,
+            source_format=self.source_format,
+            dataset_description=self.dataset_description,
+        )
+
+    def cleanup(self, release: ThothRelease, **kwargs) -> None:
+        """Delete all files and folders associated with this release."""
+        for path in [self.download_folder, self.transform_folder]:
+            try:
+                shutil.rmtree(path)
+            except FileNotFoundError as e:
+                logging.warning(f"No such file or directory {path}: {e}")
```

Comment:
It would be useful to move the cleanup functionality into a reusable function, as this will be reused across workflows, see below for function. Typically the cleanup should remove the workflow folder and also remove any XComs.

```suggestion
        """Delete all files and folders associated with this release."""
        
        cleanup(self.workflow_folder, self.dag_id, kwargs["execution_date"])
```